modules/setting/setting.go

```diff
@@ -907,6 +910,14 @@ func NewContext() {
 		}
 	}
 
+	if RunUser == "root" {
+		if !unsafeAllowRunAsRoot {
+			// Special thanks to VLC which inspired the wording of this messaging.
+			log.Fatal("Gitea is not supposed to be run as root. Sorry. If you need to use privileged TCP ports please instead use setcap and the `cap_net_bind_service` permission")
```

Comment:
When we enter this if, I think we can shutdown Gitea immediately here as otherwise Gitea could still be run as root.
```suggestion
			log.Fatal("Gitea is not supposed to be run as root. Sorry. If you need to use privileged TCP ports please instead use setcap and the `cap_net_bind_service` permission")
			os.Exit(10)
```

Reply:
log.Fatal does call os.Exit by design

Reply:
I thought `Fatal` will exit Gitea with status 1 already?

Reply:
Okay, I didn't know that. I did not expect a method called `log.Fatal` to automatically shutdown the program.
But I can see why it does that.